internal/protocol/message.go

```diff
@@ -229,12 +217,46 @@ func (m *Message) putNamedValues(values NamedValues) {
 			panic("unsupported value type")
 		}
 	}
+}
+
+// Encode the given driver values as binding parameters.
+func (m *Message) putNamedValues(values NamedValues) {
+	l := len(values)
+	if l > 255 {
+		// safeguard, should have been checked beforehand.
+		panic("too many parameters")
+	}
+
+	n := uint8(l) // N of params
+	if n == 0 {
+		return
+	}
+
+	m.putUint8(n)
+	m.putNamedValuesInner(values)
+}
+
+// Encode the given driver values as binding parameters, with a 32-bit
+// parameter count (new format).
+func (m *Message) putNamedValues32(values NamedValues) {
+	l := len(values)
+	if l > (1<<32) - 1 {
```

Comment:
Would it be clearer to use `math.MaxUint32` here? Admittedly this would feel strange